### PR TITLE
chore(core): move isNxGenerator out of workspaces

### PR DESCRIPTION
**Pull request by xiongemi**
## Related Issue(s)

Fixes #

### CODE REVIEW

File: packages/nx/src/command-line/generate/generator-utils.ts [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+  resolvedCollectionName: string;
+  normalizedGeneratorName: string;
+  schema: any;
+  implementationFactory: () => Generator<unknown>;
+  isNgCompat: boolean;
+  isNxGenerator: boolean;
+  /**
+   * @deprecated(v16): This will be removed in v16, use generatorConfiguration.aliases
+   */
+  aliases: string[];
```
Comment (FrozenPandaz) [inline packages/nx/src/command-line/generate/generator-utils.ts:29]: I think we can remove this?

File: packages/nx/src/adapter/ngcli-adapter.ts [hunk 2]
```diff
@@ -123,7 +120,14 @@ export async function scheduleTarget(
...
           nodeModule,
-          this.workspaces['resolvePaths'].bind(this.workspaces)()
+          this.root
+            ? [
+                this.root,
+                __dirname,
+                process.cwd(),
+                ...getNxRequirePaths(this.root),
+              ]
+            : [__dirname, process.cwd(), ...getNxRequirePaths()]
```
Comment (FrozenPandaz) [inline packages/nx/src/adapter/ngcli-adapter.ts:130]: What's going on here? This doesn't seem like the same as the `resolvePaths` method in `Workspaces`.

`Workspaces` was originally instantiated with `this.root` here.

So the logic should be the following `this.root ? [this.root, __dirname] : [__dirname]` right?

I mean `__dirname` is not exactly the same... but i'm not sure if it matters :thinking: 

Whether we resolve from `packages/nx/src/config` (`__dirname` of `Workspaces`) or `packages/nx/src/adapter` (`__dirname`, of `ngcli-adapter`), I don't think makes much of a difference.

**Review by FrozenPandaz — CHANGES_REQUESTED** [review]

File: packages/nx/src/adapter/ngcli-adapter.ts (hunk 2 above)
Comment (xiongemi) [reply packages/nx/src/adapter/ngcli-adapter.ts:130]: how about `this.root ? getNxRequirePaths(this.root) : getNxRequirePaths()`? if not set, readPluginPackageJson will use getNxRequirePaths as default is not passed in

File: packages/nx/src/command-line/run/executor-utils.ts [hunk 3]
```diff
@@ -96,9 +96,7 @@ function readExecutorsJson(
 } {
   const { json: packageJson, path: packageJsonPath } = readPluginPackageJson(
     nodeModule,
-    root
-      ? [root, __dirname, process.cwd(), ...getNxRequirePaths()]
-      : [__dirname, process.cwd(), ...getNxRequirePaths()]
+    root ? [root, __dirname] : [__dirname]
```
Comment (FrozenPandaz) [inline packages/nx/src/command-line/run/executor-utils.ts:99]: This one should not have changed.

**Review by FrozenPandaz — CHANGES_REQUESTED** [review]

File: packages/nx/src/command-line/run/executor-utils.ts (hunk 3 above)
Comment (xiongemi) [reply packages/nx/src/command-line/run/executor-utils.ts]: revert

**Review by FrozenPandaz — APPROVED** [review]

**Pull request merged**